test/wh_test_clientserver.c

```diff
@@ -168,66 +169,82 @@ int whTest_ClientServerSequential(void)
-        whNvmMetadata meta[1] = {{
-                .id = counter + 20,
-                .access = WOLFHSM_NVM_ACCESS_ANY,
-                .flags = WOLFHSM_NVM_FLAGS_ANY,
-                .len = 0,
-        }};
-        sprintf(meta->label, "Label:%d", meta->id);
-        sprintf(send_buffer, "Data:%d Counter:%d", meta->id, counter);
+        whNvmId id = counter + 20;
+        whNvmAccess access = WOLFHSM_NVM_ACCESS_ANY;
+        whNvmFlags flags = WOLFHSM_NVM_FLAGS_ANY;
+        whNvmSize label_len = 0;
+        char label[WOLFHSM_NVM_LABEL_LEN] = {0};
+        whNvmSize len = 0;
+
+        label_len = sprintf(label, "Label:%d", id);
+        len = sprintf(send_buffer, "Data:%d Counter:%d", id, counter);
+
+#if defined(WH_CFG_TEST_VERBOSE)
+        printf("Client NvmAddObjectRequest:%d, id:%u, access:0x%x, flags:0x%x, len:%u label:%s\nData:%s\n",
+            ret, id, access, flags, len, label, send_buffer);
+#endif
 
         WH_TEST_RETURN_ON_FAIL(
-                wh_Client_NvmGetAvailableRequest(client));
+                wh_Client_NvmAddObjectRequest(client,
+                        id, access, flags,
+                        label_len, (uint8_t*)label,
+                        len, (uint8_t*)send_buffer));
         WH_TEST_RETURN_ON_FAIL(
                 wh_Server_HandleRequestMessage(server));
         WH_TEST_RETURN_ON_FAIL(
-                wh_Client_NvmGetAvailableResponse(client,
-                        &avail_size, &avail_objects,
-                        &reclaim_size, &reclaim_objects));
+                wh_Client_NvmAddObjectResponse(client, &server_rc));
         #if defined(WH_CFG_TEST_VERBOSE)
-                printf("Client NvmGetAvailableResponse:%d, avail_size:%d avail_objects:%d, reclaim_size:%d reclaim_objects:%d\n",
-                    ret, avail_size, avail_objects, reclaim_size, reclaim_objects);
+                printf("Client NvmAddObjectResponse:%d, server_rc:%d\n",
+                    ret, server_rc);
         #endif
 
         WH_TEST_RETURN_ON_FAIL(
                 wh_Client_NvmGetAvailableRequest(client));
         WH_TEST_RETURN_ON_FAIL(
                 wh_Server_HandleRequestMessage(server));
         WH_TEST_RETURN_ON_FAIL(
-                wh_Client_NvmGetAvailableResponse(client,
+                wh_Client_NvmGetAvailableResponse(client, &server_rc,
                         &avail_size, &avail_objects,
                         &reclaim_size, &reclaim_objects));
         #if defined(WH_CFG_TEST_VERBOSE)
-                printf("Client NvmGetAvailableResponse:%d, avail_size:%d avail_objects:%d, reclaim_size:%d reclaim_objects:%d\n",
-                    ret, avail_size, avail_objects, reclaim_size, reclaim_objects);
+                printf("Client NvmGetAvailableResponse:%d, server_rc:%d, avail_size:%d avail_objects:%d, reclaim_size:%d reclaim_objects:%d\n",
+                    ret, server_rc, avail_size, avail_objects, reclaim_size, reclaim_objects);
         #endif
```

Comment:
We should probably make this portion of the test self-checking (e.g. how we did the `addObjectWithReadbackCheck` functions for the NVM flash layer) - that way the test will fail if the objects we expected to read back aren't the ones we wrote (or don't exist). Currently this requires us to read the output of printfs to sanity check everything except return values. Very easy for a broken implementation to return success but not do what we think it is doing.